paddle/fluid/framework/ir/layer_norm_fuse_pass_tester.cc

```diff
@@ -0,0 +1,293 @@
+#include "paddle/fluid/framework/program_desc.h"
+#include "paddle/fluid/framework/scope.h"
+#include "paddle/fluid/platform/errors.h"
+#include "paddle/fluid/platform/place.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+namespace {
+
+ProgramDesc BuildGraphProgram() {
+  auto prog = test::BuildProgramDesc(
+      {"x", "x_mean_out", "x_sub_mean_out", "x_sub_mean_sqr_out", "std_dev_out",
+       "std_dev_eps_out", "std_dev_eps_sqrt_out", "division_out", "scale_out",
+       "shift_out"},
+      {"sqr_pow", "eps", "gamma", "beta"});
+
+  auto* block_desc = prog.MutableBlock(0);
+  auto* x_var_desc = block_desc->Var("x");
+  x_var_desc->SetDataType(proto::VarType::FP32);
+  x_var_desc->SetShape({3, 32, 48});
+
+  auto* eps_var_desc = block_desc->Var("eps");
+  eps_var_desc->SetDataType(proto::VarType::FP32);
+  eps_var_desc->SetShape({1});
+
+  auto* gamma_var_desc = block_desc->Var("gamma");
+  gamma_var_desc->SetDataType(proto::VarType::FP32);
+  gamma_var_desc->SetShape({48});
+
+  auto* beta_var_desc = block_desc->Var("beta");
+  beta_var_desc->SetDataType(proto::VarType::FP32);
+  beta_var_desc->SetShape({48});
+
+  test::CreateOp(&prog, "reduce_mean", {{"X", "x"}}, {{"Out", "x_mean_out"}},
+                 false);
+  test::CreateOp(&prog, "elementwise_sub", {{"X", "x"}, {"Y", "x_mean_out"}},
+                 {{"Out", "x_sub_mean_out"}}, false);
+  test::CreateOp(&prog, "elementwise_pow",
+                 {{"X", "x_sub_mean_out"}, {"Y", "sqr_pow"}},
+                 {{"Out", "x_sub_mean_sqr_out"}}, false);
+  test::CreateOp(&prog, "reduce_mean", {{"X", "x_sub_mean_sqr_out"}},
+                 {{"Out", "std_dev_out"}}, false);
+  test::CreateOp(&prog, "elementwise_add", {{"X", "std_dev_out"}, {"Y", "eps"}},
+                 {{"Out", "std_dev_eps_out"}}, false);
+  test::CreateOp(&prog, "sqrt", {{"X", "std_dev_eps_out"}},
+                 {{"Out", "std_dev_eps_sqrt_out"}}, false);
+  test::CreateOp(&prog, "elementwise_div",
+                 {{"X", "x_sub_mean_out"}, {"Y", "std_dev_eps_sqrt_out"}},
+                 {{"Out", "division_out"}}, false);
+  test::CreateOp(&prog, "elementwise_mul",
+                 {{"X", "division_out"}, {"Y", "gamma"}},
+                 {{"Out", "scale_out"}}, false);
+  test::CreateOp(&prog, "elementwise_add", {{"X", "scale_out"}, {"Y", "beta"}},
+                 {{"Out", "shift_out"}}, false);
+  return prog;
+}
+
+bool CheckFusedSubgraphOpsCount(const Graph& graph) {
+  return test::AssertOpsCount(graph, {{"reduce_mean", 0},
+                                      {"elementwise_sub", 0},
+                                      {"elementwise_pow", 0},
+                                      {"elementwise_add", 0},
+                                      {"sqrt", 0},
+                                      {"elementwise_div", 0},
+                                      {"elementwise_mul", 0},
+                                      {"layer_norm", 1}});
+}
```

Comment:
Here also should be
```
  return test::AssertOpsCount(graph, {{"reduce_mean", 0},
                                      {"elementwise_sub", 1},
                                      {"elementwise_pow", 1},
                                      {"elementwise_add", 1},
                                      {"sqrt", 0},
                                      {"elementwise_div", 1},
                                      {"elementwise_mul", 1},
                                      {"layer_norm", 0}});
```
layer_norm version should be 0, because you did add checkpoint like this for it, so version 0.
```
REGISTER_OP_VERSION(elementwise_pow)
    .AddCheckpoint(
        R"ROC(Register elementwise_pow for adding the attribute of Scale_y)ROC",
        paddle::framework::compatible::OpVersionDesc().NewAttr(
```